transactions.js

```diff
@@ -0,0 +1,26 @@
+if (!isLoggedIn()) {
+  location.href = "index.html";
+}
+console.log("You can see this if only you're logged in!");
+
+const userAccountNumberElement = document.getElementById("user-account-number");
+
+const userAccountBalanceElement = document.getElementById(
+  "user-account-balance"
+);
+const buttonElem = document.getElementById("clear-history-btn");
+
+const tableDataElem = document.querySelectorAll(".transaction-data");
+
+const accountNumber = localStorage.getItem("MB_LOGGEDIN_USER_ACCOUNT_NUMBER");
+
+userAccountNumberElement.textContent = `Account: ${accountNumber}`;
+userAccountBalanceElement.textContent = `#Balance`;
+
+function deleteTransactionHistory() {
```

Comment:
The logic in the function is not correct. You shouldn't be deleting the cells, instead you should be deleting the rows. Even then, that's not the right thing to do. Merely doing this would be temporarily removing the data from the user's view.
The data has to be cleared from the database (local storage). You need to target the user in `MB_USER_ACCOUNTS` and clear his transactions.

Suggestion: 

```
// get all users
const allUsers = getAllUsers();

// find the index of the currently logged in user in our users list
const currentUserIndex = allUsers.findIndex(user => user.accountNumber === accountNumber);

// clear the current user's transactions
allUsers[currentUserIndex].transactions = [];
```

After removing this from local storage, you need to refresh the table. To achieve this, you need to put the logic that displays the data in the table inside a function. All you need to do is call the function again after clearing the transactions in local storage.

Reply:
Also give the function a more relevant name.
Suggested: `clearTransactionsHistory`

Reply:
Done